accounts/abi/bind/bindv2.go

```diff
@@ -0,0 +1,311 @@
+}
+
+// RegisterErrorIdentifier applies registerIdentifier for contract errors.
+func (b *contractBinder) RegisterErrorIdentifier(id string) (string, error) {
+	return b.registerIdentifier(b.errorIdentifiers, id)
+}
+
+// BindStructType register the type to be emitted as a struct in the
+// bindings.
+func (b *binder) BindStructType(typ abi.Type) {
+	bindStructType(typ, b.structs)
+}
+
+// contractBinder holds state for binding of a single contract
+type contractBinder struct {
+	binder *binder
+	calls  map[string]*tmplMethod
+	events map[string]*tmplEvent
+	errors map[string]*tmplError
+
+	callIdentifiers  map[string]bool
+	eventIdentifiers map[string]bool
+	errorIdentifiers map[string]bool
+}
+
+// bindMethod registers a method to be emitted in the bindings.
+// The name, inputs and outputs are normalized.  If any inputs are
+// struct-type their structs are registered to be emitted in the bindings.
+// Any methods that return more than one output have their result coalesced
+// into a struct.
+func (cb *contractBinder) bindMethod(original abi.Method) error {
+	normalized := original
+	normalizedName, err := cb.RegisterCallIdentifier(original.Name)
+	if err != nil {
+		return err
+	}
+
+	normalized.Name = normalizedName
+	normalized.Inputs = make([]abi.Argument, len(original.Inputs))
+	copy(normalized.Inputs, original.Inputs)
+	for j, input := range normalized.Inputs {
+		if input.Name == "" || isKeyWord(input.Name) {
+			normalized.Inputs[j].Name = fmt.Sprintf("arg%d", j)
+		}
+		if hasStruct(input.Type) {
+			cb.binder.BindStructType(input.Type)
+		}
+	}
+	normalized.Outputs = make([]abi.Argument, len(original.Outputs))
+	copy(normalized.Outputs, original.Outputs)
+	for j, output := range normalized.Outputs {
+		if output.Name != "" {
+			normalized.Outputs[j].Name = capitalise(output.Name)
+		}
+		if hasStruct(output.Type) {
+			cb.binder.BindStructType(output.Type)
+		}
+	}
+	isStructured := structured(original.Outputs)
+	// if the call returns multiple values, coallesce them into a struct
+	if len(normalized.Outputs) > 1 {
+		// Build up dictionary of existing arg names.
+		keys := make(map[string]struct{})
+		for _, o := range normalized.Outputs {
+			if o.Name != "" {
+				keys[strings.ToLower(o.Name)] = struct{}{}
+			}
+		}
+		// Assign names to anonymous fields.
+		for i, o := range normalized.Outputs {
+			if o.Name != "" {
+				continue
+			}
+			o.Name = capitalise(abi.ResolveNameConflict("arg", func(name string) bool { _, ok := keys[name]; return ok }))
+			normalized.Outputs[i] = o
+			keys[strings.ToLower(o.Name)] = struct{}{}
+		}
+		isStructured = true
+	}
+
+	cb.calls[original.Name] = &tmplMethod{Original: original, Normalized: normalized, Structured: isStructured}
+	return nil
+}
+
+// normalizeErrorOrEventFields normalizes errors/events for emitting through bindings:
+// Any anonymous fields are given generated names.
+func (cb *contractBinder) normalizeErrorOrEventFields(originalInputs abi.Arguments) abi.Arguments {
```

Comment:
This method does multiple things. Howver, it only uses `cb` in order to do this: 
```
		if hasStruct(input.Type) {
			cb.binder.BindStructType(input.Type)
		}
```
Hence, would it be possible to change into 
```golang
// normalizeErrorOrEventFields normalizes errors/events for emitting through bindings:
// Any anonymous fields are given generated names.
func (cb *contractBinder) normalizeErrorOrEventFields(originalInputs abi.Arguments) abi.Arguments {
	args := normalizeErrorOrEventFields(originalInputs)
	// Soup up the struct types
	for _, input := range args {
		if hasStruct(input.Type) {
			cb.binder.BindStructType(input.Type)
		}
	}
	return args
}

func normalizeErrorOrEventFields(originalInputs abi.Arguments) abi.Arguments {
...
```
? 
Would make it more testable